dataflowjobstatus sensor dag example fixed for issue #17180

orossini opened the pull request:
<!--
Thank you for contributing! Please make sure that your code changes
are covered with tests. And in case of new features or big changes
remember to adjust the documentation.

Feel free to ping committers for the review!

In case of existing issue, reference it using one of the following:

closes: #ISSUE
related: #ISSUE

How to write a good git commit message:
http://chris.beams.io/posts/git-commit/
-->


---
**^ Add meaningful description above**

closes issue #17180

Update to the DAG examples for DataflowJobStatusSensor to avoid error.
The current example pulls XCOM to get Dataflow job id from XCOM attributes.
However an incorrect attribute name is used in the current DAG example.

the XCOM doesn't have any attribute named "job_id" when using :
job_id="{{task_instance.xcom_pull('start-python-job-async')['job_id']}}",
instead "id" has to be used to get the related Dataflow job id.

orossini committed:
dataflowjobstatus sensor dag example updated

mik-laj commented on airflow/providers/google/cloud/example_dags/example_dataflow.py:
```diff
@@ -166,7 +166,7 @@
     wait_for_python_job_async_done = DataflowJobStatusSensor(
         task_id="wait-for-python-job-async-done",
-        job_id="{{task_instance.xcom_pull('start-python-job-async')['job_id']}}",
+        job_id="{{task_instance.xcom_pull('start-python-job-async')['id']}}",
```
Are you sure? Please look at https://github.com/apache/airflow/blob/82de4a87b2a36e88e0b5803ed85e17d30719819a/airflow/providers/google/cloud/operators/dataflow.py#L1150
Did I miss something?

orossini replied:
You're right. However there is an issue for DataflowTemplatedJobStartOperator. The XCom doesn't have any job_id attribute.
To solve this, the operators/dataflow.py can be updated for DataflowTemplatedJobStartOperator (as well as flextemplateJob and sqlJob operators): return job_id instead of job.

orossini committed:
dataflowjobstatus sensor dag example updated

updated Operators return value to get job_id in XCom for sqlJob, templateJob and flexTemplateJob

orossini committed:
dataflowjobstatus sensor dag example reverted to previous state
